README.md

```diff
@@ -2,3 +2,41 @@
 
 This is the new [Flask](http://flask.pocoo.org/) application that will gradually replace the website at https://snapcraft.io, starting with the snap details page.
 
+## Local development
+
+To run the site for local development, follow these steps:
+
+### Installing dependencies
+
+``` bash
+python3 -m venv env3             # Create a python3 virtual environment
+source env3/bin/activate         # Use the python3 environment
+pip install -r requirements.txt  # Install python dependencies
+yarn install                     # Install NPM dependencies into node_modules
+```
+
+### Building CSS
+
+``` bash
+yarn run build  # Build SCSS files into CSS
+```
+
+### Run the site
+
+To run the local webserver at http://127.0.0.1:5000 :
+
+``` bash
+FLASK_APP=app.py FLASK_DEBUG=true flask run
+```
+
+### Testing
+
+First make sure the site's dependencies are installed. Then:
+
+``` bash
+pip install flake8  # Install flake8 for syntax testing
```

Comment:
Why it's not part of requirements.txt?

Reply:
Because it's not a dependency of the site - it's only a dependency for local testing. In our standard ./run architecture, flake8 (syntax checker) and ipdb (debugging tool) come preinstalled in our python image, so you don't need to put them in site dependencies.